lib/modules/manager/gradle/extract/consistent-versions-plugin.ts

```diff
@@ -0,0 +1,212 @@
+import { newlineRegex, regEx } from '../../../../util/regex';
+import type { PackageDependency } from '../../types';
+import type { GradleManagerData } from '../types';
+import { isDependencyString, versionLikeSubstring } from '../utils';
+
+export const VERSIONS_PROPS = 'versions.props';
+export const VERSIONS_LOCK = 'versions.lock';
+const LOCKFILE_HEADER_TEXT =
+  '# Run ./gradlew --write-locks to regenerate this file';
+
+/**
+ * Determines if Palantir gradle-consistent-versions is in use, https://github.com/palantir/gradle-consistent-versions.
+ * Both `versions.props` and `versions.lock` must exist and the special header line of lock file must match
+ *
+ * @param versionsPropsFilename is the full file name path of `versions.props`
+ * @param fileContents map with file contents of all files
+ */
+export function usesGcv(
+  versionsPropsFilename: string,
+  fileContents: Record<string, string | null>
+): boolean {
+  const versionsLockFile: string = fs.getSiblingFileName(
+    versionsPropsFilename,
+    VERSIONS_LOCK
+  );
+  return (
+    fileContents[versionsLockFile]?.startsWith(LOCKFILE_HEADER_TEXT) ?? false
+  );
+}
+
+/**
+ * Confirms whether the provided file name is the props file
+ */
+export function isGcvPropsFile(fileName: string): boolean {
+  return fileName === VERSIONS_PROPS || fileName.endsWith(`/${VERSIONS_PROPS}`);
+}
+
+/**
+ * Confirms whether the provided file name is the lock file
+ */
+export function isGcvLockFile(fileName: string): boolean {
+  return fileName === VERSIONS_LOCK || fileName.endsWith(`/${VERSIONS_LOCK}`);
+}
+
+/**
+ * Parses Gradle-Consistent-Versions files to figure out what dependencies, versions
+ * and groups they contain. The parsing goes like this:
+ * - Parse `versions.props` into deps (or groups) and versions, remembering file offsets
+ * - Parse `versions.lock` into deps and lock-versions
+ * - For each exact dep in props file, lookup the lock-version from lock file
+ * - For each group/regex dep in props file, lookup the set of exact deps and versions in lock file
+ *
+ * @param propsFileName name and path of the props file
+ * @param fileContents text content of all files
+ */
+export function parseGcv(
+  propsFileName: string,
+  fileContents: Record<string, string | null>
+): PackageDependency<GradleManagerData>[] {
+  const propsFileContent = fileContents[propsFileName] ?? '';
+  const lockFileName = fs.getSiblingFileName(propsFileName, VERSIONS_LOCK);
+  const lockFileContent = fileContents[lockFileName] ?? '';
+  const lockFileMap = parseLockFile(lockFileContent);
+  const [propsFileExactMap, propsFileRegexMap] =
+    parsePropsFile(propsFileContent);
+
+  const extractedDeps: PackageDependency<GradleManagerData>[] = [];
+
+  // For each exact dep in props file
+  for (const [propDep, versionAndPosition] of propsFileExactMap) {
+    if (lockFileMap.has(propDep)) {
+      const newDep: Record<string, any> = {
+        managerData: {
+          packageFile: VERSIONS_PROPS,
```

Comment:
This is a bug - packageFile must be the one we receive, else sub-folders won't work

Reply:
Fixed